tasks/eth/base-002-fp-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,296 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {NestedSignFromJson as OriginalNestedSignFromJson} from "script/NestedSignFromJson.s.sol";
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {L1StandardBridge} from "@eth-optimism-bedrock/src/L1/L1StandardBridge.sol";
+import {ProtocolVersion, ProtocolVersions} from "@eth-optimism-bedrock/src/L1/ProtocolVersions.sol";
+import {SuperchainConfig} from "@eth-optimism-bedrock/src/L1/SuperchainConfig.sol";
+import {OptimismPortal2} from "@eth-optimism-bedrock/src/L1/OptimismPortal2.sol";
+import {L1CrossDomainMessenger} from "@eth-optimism-bedrock/src/L1/L1CrossDomainMessenger.sol";
+import {OptimismMintableERC20Factory} from "@eth-optimism-bedrock/src/universal/OptimismMintableERC20Factory.sol";
+import {L1ERC721Bridge} from "@eth-optimism-bedrock/src/L1/L1ERC721Bridge.sol";
+import {AddressManager} from "@eth-optimism-bedrock/src/legacy/AddressManager.sol";
+import {Predeploys} from "@eth-optimism-bedrock/src/libraries/Predeploys.sol";
+import {ResourceMetering} from "@eth-optimism-bedrock/src/libraries/Constants.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {ISemver} from "@eth-optimism-bedrock/src/universal/ISemver.sol";
+import {EIP1967Helper} from "@eth-optimism-bedrock/test/mocks/EIP1967Helper.sol";
+import {IGnosisSafe, Enum} from "@eth-optimism-bedrock/scripts/interfaces/IGnosisSafe.sol";
+import {GnosisSafe} from "safe-contracts/GnosisSafe.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {Constants} from "@eth-optimism-bedrock/src/libraries/Constants.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {IMulticall3} from "forge-std/interfaces/IMulticall3.sol";
+import {LibString} from "solady/utils/LibString.sol";
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import "@eth-optimism-bedrock/src/dispute/lib/Types.sol";
+import {stdToml} from "forge-std/StdToml.sol";
+
+contract NestedSignFromJson is OriginalNestedSignFromJson {
+    using LibString for string;
+
+    // Safe contract for this task.
+    GnosisSafe securityCouncilSafe = GnosisSafe(payable(vm.envAddress("COUNCIL_SAFE")));
+    GnosisSafe fndSafe = GnosisSafe(payable(vm.envAddress("FOUNDATION_SAFE")));
+    GnosisSafe ownerSafe = GnosisSafe(payable(vm.envAddress("OWNER_SAFE")));
```

Comment:
In `setUp`, we should have safety check that calls `address[] memory owners = ownerSafe.getOwners()` and asserts that:
- the length is 2
- one signer is the `baseSafe`, and the other signer is the `foundationSafe`
This helps automate the verification that we have all the right env vars